benchmark/Cpp/Savina/src/concurrency/Banking.lf

```diff
@@ -0,0 +1,201 @@
+            size_t src_account = randomGen.nextInt((numAccounts / 10) * 8);
+            size_t loop_id = randomGen.nextInt(numAccounts - src_account);
+            if(loop_id == 0) {
+                loop_id += 1;
+            }
+            size_t dest_account = src_account + loop_id;
+            
+            double amount = randomGen.nextDouble() * 1000;
+            
+            reactor::log::Info() << "Send " << amount << " from " << src_account << " to " << dest_account;
+            messageQueues[src_account].push_back(CreditMessage{dest_account, amount});
+        }
+    =}
+    
+    reaction(start) -> reset, next {=
+        reactor::log::Info() << "Teller: Start a new iteration";
+        
+        // reset local state
+        randomGen = PseudoRandom(123456);
+
+        // reset all accounts
+        for (auto& r : reset)
+            r.set();
+
+        generateWork();
+                
+        // start execution
+        next.schedule();
+    =}
+    
+    reaction(next) -> next, credit, finished {=
+        reactor::log::Info() << "Teller: Sending a new round of credit messages";
+        bool work_found{false};
+        for (size_t i{0}; i < numAccounts; i++) {
+            auto& queue = messageQueues[i];
+            if (!queue.empty()) {
+                work_found = true;
+                const auto& message = queue.front();  
+                credit[message.recipient].set(message);
+                queue.pop_front();
+            }
+        }
+        
+        if (work_found) {
+            next.schedule();
+        } else {
+            reactor::log::Info() << "Teller: Finished iteration";
+            finished.set();
+        }
+    =}
+}
+
+reactor Account(bank_index:size_t(0), numAccounts:size_t(1000), initialBalance:double(0.0)) {
+    state balance: double{0.0};
+    
+    input reset: void; 
+    input inCredit: CreditMessage;
+   
+    input[numAccounts] inDebit: double; 
+    output[numAccounts] outDebit: double;
+       
+    reaction (reset) {=
+        balance = initialBalance;
+    =}
+    
+    reaction (inCredit) -> outDebit {=
+        auto message = inCredit.get();
+        // reduce the balance
+        balance -= message->amount;
+        // and sent the recipient a debit message
+        outDebit[message->recipient].set(message->amount);
+        reactor::log::Info() << "Account " << bank_index << " credits " << message->amount 
+                             << " to " << message->recipient;
+    =}
+    
+    reaction (inDebit) {=
+        for (auto& port : inDebit) {
+            if (port.is_present()) {
+                double amount = *port.get();
+                // increase the balance
+                balance += amount;
+                reactor::log::Info() << "Account " << bank_index << " received " << amount;                
+            }
+        }
+    =}
+}
+
+// FIXME default value of numAccounts should be 1000
```

Comment:
Why this FIXME?

Reply:
Due to #433 we cannot set the number to 1000. Otherwise, Epoch would crash when opening the file. I will add this to the FIXME.